.github/workflows/rust.yml

```diff
@@ -505,12 +505,8 @@ jobs:
 
       - name: Check Cargo.toml formatting
         run: |
-          # if you encounter error, try rerun the command below, finally run 'git diff' to
-          # check which Cargo.toml introduces formatting violation
-          #
           # ignore ./Cargo.toml because putting workspaces in multi-line lists make it easy to read
```

Comment:
Can you please leave comments here about how to fix the issue if the CI fails when someone didn't run rust_fmt_toml correctly?

When I purposely added a problem to `Cargo.toml` I got an error that I would find hard to interpret

```shell
alamb@MacBook-Pro-8:~/Software/arrow-datafusion2$ if `./ci/scripts/rust_toml_fmt.sh`; then echo "pass" ; else  echo "failed"; fi
....
+ cargo tomlfmt -d -p ./datafusion/sqllogictest/Cargo.toml
 WARN 2023-10-09T21:27:06Z: cargo_tomlfmt: dryrun founds problems in Cargo.toml
 WARN 2023-10-09T21:27:06Z: cargo_tomlfmt: exit with -1
failed
```

Reply:
I think the reason  CI originally ran`rust_fmt_toml.sh`   + `git diff --exit-code` was so that people could run `rust_fmt_toml.sh` locally and check in the result to fix CI. Perhaps we can restore that behavior